### PR TITLE
hip@5.2.0 onwards: set prefix properly

### LINKED ISSUE #32826: petsc@3.17.4 +rocm configure fails using ROCm 5.2.0 and PrgEnv-gnu/cray/amd on OLCF Crusher
### Steps to reproduce the issue

`petsc@3.17.4 +rocm` is not building with any of the following PrgEnv on Crusher:
* `PrgEnv-gnu`
* `PrgEnv-cray`
* `PrgEnv-amd`

In all cases, I'm trying to use `rocm/5.2.0` defined as externals.

<details><summary>Spack environment</summary><pre>
spack:
  view: false
  concretizer:
    reuse: false
    unify: false
  compilers:
  - compiler:
      spec: gcc@11.2.0
      paths:
        cc: gcc
        cxx: g++
        f77: gfortran
        fc: gfortran
      flags: {}
      operating_system: sles15
      target: any
      modules:
      - PrgEnv-gnu
      - gcc/11.2.0
      environment: {}
      extra_rpaths: []
  packages:
    all:
      compiler: [gcc@11.2.0]
      providers:
        blas: [openblas]
        lapack: [openblas]
        mpi: [cray-mpich]
        scalapack: [netlib-scalapack]
      target: [zen3]
      variants: +mpi
    binutils:
      variants: +ld +gold +headers +libiberty ~nls
    elfutils:
      variants: +bzip2 ~nls +xz
    hdf5:
      variants: +fortran +hl +shared
    libfabric:
      buildable: false
      externals:
      - spec: libfabric@1.15
        modules:
        - libfabric/1.15.0.0

    # ROCm 5.2.0
    libunwind:
      variants: +pic +xz
    mpich:
      variants: ~wrapperrpath
    ncurses:
      variants: +termlib
    openblas:
      variants: threads=openmp
    python:
      version: [3.8.13]
    trilinos:
      variants: +amesos +amesos2 +anasazi +aztec +belos +boost +epetra +epetraext
        +ifpack +ifpack2 +intrepid +intrepid2 +isorropia +kokkos +ml +minitensor +muelu
        +nox +piro +phalanx +rol +rythmos +sacado +stk +shards +shylu +stokhos +stratimikos
        +teko +tempus +tpetra +trilinoscouplings +zoltan +zoltan2 +superlu-dist gotype=long_long
    xz:
      variants: +pic
    mesa:
      version: [21.3.8]
    petsc:
      variants: +batch
    parallel-netcdf:
      variants: +cxx +fortran +fpic
    netlib-scalapack:
      variants: +fpic
    netcdf-c:
      variants: ~hdf4 +mpi +parallel-netcdf +shared
    fftw:
      variants: precision=float,double,long_double
    darshan-runtime:
      variants: scheduler=slurm
    cray-mpich:
      buildable: false
      externals:
      - spec: cray-mpich@8.1.16 %gcc@11.2.0
        prefix: /opt/cray/pe/mpich/8.1.16/ofi/gnu/9.1
        modules:
        - cray-mpich/8.1.16
        - libfabric/1.15.0.0
    comgr:
      buildable: false
      externals:
      - spec: comgr@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    hip-rocclr:
      buildable: false
      externals:
      - spec: hip-rocclr@5.2.0
        prefix: /opt/rocm-5.2.0/hip
        modules: [rocm/5.2.0]
    hipblas:
      buildable: false
      externals:
      - spec: hipblas@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    hipcub:
      buildable: false
      externals:
      - spec: hipcub@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    hipfft:
      buildable: false
      externals:
      - spec: hipfft@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    hipsparse:
      buildable: false
      externals:
      - spec: hipsparse@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    miopen-hip:
      buildable: false
      externals:
      - spec: hip-rocclr@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    miopengemm:
      buildable: false
      externals:
      - spec: miopengemm@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    rccl:
      buildable: false
      externals:
      - spec: rccl@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    rocblas:
      buildable: false
      externals:
      - spec: rocblas@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    rocfft:
      buildable: false
      externals:
      - spec: rocfft@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    rocm-clang-ocl:
      buildable: false
      externals:
      - spec: rocm-clang-ocl@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    rocm-cmake:
      buildable: false
      externals:
      - spec: rocm-cmake@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    rocm-dbgapi:
      buildable: false
      externals:
      - spec: rocm-dbgapi@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    rocm-debug-agent:
      buildable: false
      externals:
      - spec: rocm-debug-agent@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    rocm-device-libs:
      buildable: false
      externals:
      - spec: rocm-device-libs@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    rocm-gdb:
      buildable: false
      externals:
      - spec: rocm-gdb@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    rocm-opencl:
      buildable: false
      externals:
      - spec: rocm-opencl@5.2.0
        prefix: /opt/rocm-5.2.0/opencl
        modules: [rocm/5.2.0]
    rocm-smi-lib:
      buildable: false
      externals:
      - spec: rocm-smi-lib@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    hip:
      buildable: false
      externals:
      - spec: hip@5.2.0
        prefix: /opt/rocm-5.2.0/hip
        modules: [rocm/5.2.0]
        extra_attributes:
          compilers:
            c: /opt/rocm-5.2.0/llvm/bin/clang++
            c++: /opt/rocm-5.2.0/llvm/bin/clang++
            hip: /opt/rocm-5.2.0/hip/bin/hipcc
    llvm-amdgpu:
      buildable: false
      externals:
      - spec: llvm-amdgpu@5.2.0
        prefix: /opt/rocm-5.2.0/llvm
        modules: [rocm/5.2.0]
        extra_attributes:
          compilers:
            c: /opt/rocm-5.2.0/llvm/bin/clang++
            cxx: /opt/rocm-5.2.0/llvm/bin/clang++
    hsakmt-roct:
      buildable: false
      externals:
      - spec: hsakmt-roct@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    hsa-rocr-dev:
      buildable: false
      externals:
      - spec: hsa-rocr-dev@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
        extra_atributes:
          compilers:
            c: /opt/rocm-5.2.0/llvm/bin/clang++
            cxx: /opt/rocm-5.2.0/llvm/bin/clang++
    roctracer-dev-api:
      buildable: false
      externals:
      - spec: roctracer-dev-api@5.2.0
        prefix: /opt/rocm-5.2.0/roctracer
        modules: [rocm/5.2.0]
    rocprim:
      buildable: false
      externals:
      - spec: rocprim@5.2.0
        prefix: /opt/rocm-5.2.0
        modules: [rocm/5.2.0]
    rocrand:
      buildable: false
      externals:
      - spec: rocrand@5.2.0
        prefix: /opt/rocm-5.2.0
        modules: [rocm/5.2.0]
    rocsolver:
      buildable: false
      externals:
      - spec: rocsolver@5.2.0
        prefix: /opt/rocm-5.2.0
        modules: [rocm/5.2.0]
    rocsparse:
      buildable: false
      externals:
      - spec: rocsparse@5.2.0
        prefix: /opt/rocm-5.2.0
        modules: [rocm/5.2.0]
    rocthrust:
      buildable: false
      externals:
      - spec: rocthrust@5.2.0
        prefix: /opt/rocm-5.2.0
        modules: [rocm/5.2.0]
  specs:
  - petsc@3.17.4 +rocm amdgpu_target=gfx90a                   # petsc
</pre></details>

<details><summary>Concretization</summary><pre>
 -   z3h5sxp  petsc@3.17.4%gcc@11.2.0~X+batch~cgns~complex~cuda~debug+double~exodusii~fftw+fortran~giflib+hdf5~hpddm~hwloc+hypre~int64~jpeg~knl~kokkos~libpng~libyaml~memkind+metis~mkl-pardiso~mmg~moab~mpfr+mpi~mumps~openmp~p4est~parmmg~ptscotch~random123+rocm~saws~scalapack+shared~strumpack~suite-sparse+superlu-dist~tetgen~trilinos~valgrind amdgpu_target=gfx90a clanguage=C arch=cray-sles15-zen3
[+]  yfbzrpg      ^cray-mpich@8.1.16%gcc@11.2.0+wrappers arch=cray-sles15-zen3
[+]  3s2io65      ^diffutils@3.8%gcc@11.2.0 arch=cray-sles15-zen3
[+]  po4cjxv          ^libiconv@1.16%gcc@11.2.0 libs=shared,static arch=cray-sles15-zen3
[+]  pcmpoth      ^hdf5@1.12.2%gcc@11.2.0~cxx+fortran+hl~ipo~java+mpi+shared~szip~threadsafe+tools api=default build_type=RelWithDebInfo arch=cray-sles15-zen3
[+]  ymt6u65          ^cmake@3.23.3%gcc@11.2.0~doc+ncurses+ownlibs~qt build_type=Release arch=cray-sles15-zen3
[+]  33sp27w              ^ncurses@6.3%gcc@11.2.0~symlinks+termlib abi=none arch=cray-sles15-zen3
[+]  pysfq3q                  ^pkgconf@1.8.0%gcc@11.2.0 arch=cray-sles15-zen3
[+]  riewo2i              ^openssl@1.1.1q%gcc@11.2.0~docs~shared certs=mozilla patches=3fdcf2d arch=cray-sles15-zen3
[+]  hql4jjz                  ^ca-certificates-mozilla@2022-07-19%gcc@11.2.0 arch=cray-sles15-zen3
[+]  ks6qvkk                  ^perl@5.34.1%gcc@11.2.0+cpanm+shared+threads arch=cray-sles15-zen3
[+]  kjdztqc                      ^berkeley-db@18.1.40%gcc@11.2.0+cxx~docs+stl patches=b231fcc arch=cray-sles15-zen3
[+]  rll2prr                      ^bzip2@1.0.8%gcc@11.2.0~debug~pic+shared arch=cray-sles15-zen3
[+]  yqxusqu                      ^gdbm@1.19%gcc@11.2.0 arch=cray-sles15-zen3
[+]  fdyfuat                          ^readline@8.1.2%gcc@11.2.0 arch=cray-sles15-zen3
[+]  ymk6b64                      ^zlib@1.2.12%gcc@11.2.0+optimize+pic+shared patches=0d38234 arch=cray-sles15-zen3
[+]  7dkpqew      ^hip@5.2.0%gcc@11.2.0~ipo build_type=Release patches=959d1fe arch=cray-sles15-zen3
[+]  wybjsay      ^hipblas@5.2.0%gcc@11.2.0~ipo build_type=Release arch=cray-sles15-zen3
[+]  a6zuh3i      ^hipsparse@5.2.0%gcc@11.2.0~ipo build_type=Release arch=cray-sles15-zen3
[+]  ykiof3l      ^hsa-rocr-dev@5.2.0%gcc@11.2.0+image~ipo+shared build_type=Release patches=71e6851 arch=cray-sles15-zen3
[+]  klydyzx      ^hypre@2.25.0%gcc@11.2.0~complex~cuda~debug+fortran~gptune~int64~internal-superlu~mixedint+mpi~openmp~rocm+shared~superlu-dist~unified-memory arch=cray-sles15-zen3
[+]  dhvkyk4          ^openblas@0.3.20%gcc@11.2.0~bignuma~consistent_fpcsr~ilp64+locking+pic+shared symbol_suffix=none threads=openmp arch=cray-sles15-zen3
[+]  hngdnef      ^llvm-amdgpu@5.2.0%gcc@11.2.0~ipo~link_llvm_dylib~llvm_dylib~openmp+rocm-device-libs build_type=Release patches=a08bbe1 arch=cray-sles15-zen3
[+]  qbo2bmz      ^metis@5.1.0%gcc@11.2.0~gdb~int64~real64+shared build_type=Release patches=4991da9,b1225da arch=cray-sles15-zen3
[+]  h4ttdnr      ^parmetis@4.0.3%gcc@11.2.0~gdb~int64~ipo+shared build_type=RelWithDebInfo patches=4f89253,50ed208,704b84f arch=cray-sles15-zen3
[+]  daysjb6      ^python@3.8.13%gcc@11.2.0+bz2+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib patches=0d98e93,4c24573,ebdca64,f2fd060 arch=cray-sles15-zen3
[+]  kb64bk6          ^expat@2.4.8%gcc@11.2.0+libbsd arch=cray-sles15-zen3
[+]  6ii42bx              ^libbsd@0.11.5%gcc@11.2.0 arch=cray-sles15-zen3
[+]  nzueemp                  ^libmd@1.0.4%gcc@11.2.0 arch=cray-sles15-zen3
[+]  5nmtuqz          ^gettext@0.21%gcc@11.2.0+bzip2+curses+git~libunistring+libxml2+tar+xz arch=cray-sles15-zen3
[+]  yzo26m3              ^libxml2@2.9.13%gcc@11.2.0~python arch=cray-sles15-zen3
[+]  7n4wqcj                  ^xz@5.2.5%gcc@11.2.0+pic libs=shared,static arch=cray-sles15-zen3
[+]  pa7goe7              ^tar@1.34%gcc@11.2.0 zip=pigz arch=cray-sles15-zen3
[+]  5yysicj                  ^pigz@2.7%gcc@11.2.0 arch=cray-sles15-zen3
[+]  4cs4kky                  ^zstd@1.5.2%gcc@11.2.0+programs compression=none libs=shared,static arch=cray-sles15-zen3
[+]  46izinn          ^libffi@3.4.2%gcc@11.2.0 arch=cray-sles15-zen3
[+]  4xmsjik          ^sqlite@3.39.2%gcc@11.2.0+column_metadata+dynamic_extensions+fts~functions+rtree arch=cray-sles15-zen3
[+]  pgmt2lr          ^util-linux-uuid@2.37.4%gcc@11.2.0 arch=cray-sles15-zen3
[+]  5rahcjv      ^rocblas@5.2.0%gcc@11.2.0~ipo+tensile amdgpu_target=auto build_type=Release patches=81591d9 arch=cray-sles15-zen3
[+]  wl3jsgl      ^rocprim@5.2.0%gcc@11.2.0~ipo amdgpu_target=auto build_type=Release arch=cray-sles15-zen3
[+]  4jm3vai      ^rocrand@5.2.0%gcc@11.2.0~ipo amdgpu_target=auto build_type=Release arch=cray-sles15-zen3
[+]  3ddnqhp      ^rocsolver@5.2.0%gcc@11.2.0~ipo+optimal amdgpu_target=auto build_type=Release arch=cray-sles15-zen3
[+]  zplk7dv      ^rocsparse@5.2.0%gcc@11.2.0~ipo amdgpu_target=auto build_type=Release arch=cray-sles15-zen3
[+]  gajr6qt      ^rocthrust@5.2.0%gcc@11.2.0~ipo build_type=Release arch=cray-sles15-zen3
[+]  laqrbdx      ^superlu-dist@7.2.0%gcc@11.2.0~cuda~int64~ipo~openmp~rocm+shared build_type=RelWithDebInfo patches=8da9e21 arch=cray-sles15-zen3
</pre></details>

Error:
```
$> time spack -e . install
...
==> Installing petsc-3.17.4-z3h5sxpo2kegc6eghkrcxsgyqrbsxrvr
==> No binary for petsc-3.17.4-z3h5sxpo2kegc6eghkrcxsgyqrbsxrvr found: installing from source
==> Using cached archive: /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-gnu/spack/var/spack/cache/_source-cache/archive/99/99c127486722a3ffd95a268b4ceb0976cbf217926c681a9631bd7246eab8cb2a.tar.gz
==> No patches needed for petsc
==> petsc: Executing phase: 'configure'
==> Error: ProcessError: Command exited with status 1:
    '/gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-gnu/spack/opt/spack/cray-sles15-zen3/gcc-11.2.0/python-3.8.13-daysjb6sle5nfaftamz6j2ncocnyeaut/bin/python3.8' 'configure' '--prefix=/gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-gnu/spack/opt/spack/cray-sles15-zen3/gcc-11.2.0/petsc-3.17.4-z3h5sxpo2kegc6eghkrcxsgyqrbsxrvr' '--with-ssl=0' '--download-c2html=0' '--download-sowing=0' '--download-hwloc=0' '--with-cc=/opt/cray/pe/mpich/8.1.16/ofi/gnu/9.1/bin/mpicc' '--with-cxx=/opt/cray/pe/mpich/8.1.16/ofi/gnu/9.1/bin/mpicxx' '--with-fc=/opt/cray/pe/mpich/8.1.16/ofi/gnu/9.1/bin/mpif90' '--with-precision=double' '--with-scalar-type=real' '--with-shared-libraries=1' '--with-debugging=0' '--with-openmp=0' '--with-64-bit-indices=0' '--with-blas-lapack-lib=/gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-gnu/spack/opt/spack/cray-sles15-zen3/gcc-11.2.0/openblas-0.3.20-dhvkyk4mab26l3vsisbycqcxxu4y4ojt/lib/libopenblas.so' '--with-batch=1' '--with-x=0' '--with-clanguage=C' '--with-cuda=0' '--with-hip=1' '--with-hip-include=/opt/rocm-5.2.0/hip/include' '--with-metis=1' '--with-metis-include=/gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-gnu/spack/opt/spack/cray-sles15-zen3/gcc-11.2.0/metis-5.1.0-qbo2bmzf27ky5mxrfmksadzg7bfnknag/include' '--with-metis-lib=/gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-gnu/spack/opt/spack/cray-sles15-zen3/gcc-11.2.0/metis-5.1.0-qbo2bmzf27ky5mxrfmksadzg7bfnknag/lib/libmetis.so' '--with-hypre=1' '--with-hypre-include=/gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-gnu/spack/opt/spack/cray-sles15-zen3/gcc-11.2.0/hypre-2.25.0-klydyzxuk6l6xg4goaqvvsipctapcq2p/include' '--with-hypre-lib=/gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-gnu/spack/opt/spack/cray-sles15-zen3/gcc-11.2.0/hypre-2.25.0-klydyzxuk6l6xg4goaqvvsipctapcq2p/lib/libHYPRE.so' '--with-parmetis=1' '--with-parmetis-include=/gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-gnu/spack/opt/spack/cray-sles15-zen3/gcc-11.2.0/parmetis-4.0.3-h4ttdnr7aeui6nq5csur3ppnm5iv4f4f/include' '--with-parmetis-lib=/gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-gnu/spack/opt/spack/cray-sles15-zen3/gcc-11.2.0/parmetis-4.0.3-h4ttdnr7aeui6nq5csur3ppnm5iv4f4f/lib/libparmetis.so' '--with-kokkos=0' '--with-kokkos-kernels=0' '--with-superlu_dist=1' '--with-superlu_dist-include=/gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-gnu/spack/opt/spack/cray-sles15-zen3/gcc-11.2.0/superlu-dist-7.2.0-laqrbdxumxiomqfnzy4ul7jp65yxmdgu/include' '--with-superlu_dist-lib=/gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-gnu/spack/opt/spack/cray-sles15-zen3/gcc-11.2.0/superlu-dist-7.2.0-laqrbdxumxiomqfnzy4ul7jp65yxmdgu/lib/libsuperlu_dist.so' '--with-ptscotch=0' '--with-suitesparse=0' '--with-hdf5=1' '--with-hdf5-include=/gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-gnu/spack/opt/spack/cray-sles15-zen3/gcc-11.2.0/hdf5-1.12.2-pcmpothn4sir2fls465cojwxkkwrvdxq/include' '--with-hdf5-lib=/gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-gnu/spack/opt/spack/cray-sles15-zen3/gcc-11.2.0/hdf5-1.12.2-pcmpothn4sir2fls465cojwxkkwrvdxq/lib/libhdf5_hl_fortran.so /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-gnu/spack/opt/spack/cray-sles15-zen3/gcc-11.2.0/hdf5-1.12.2-pcmpothn4sir2fls465cojwxkkwrvdxq/lib/libhdf5_hl_f90cstub.so /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-gnu/spack/opt/spack/cray-sles15-zen3/gcc-11.2.0/hdf5-1.12.2-pcmpothn4sir2fls465cojwxkkwrvdxq/lib/libhdf5_hl.so /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-gnu/spack/opt/spack/cray-sles15-zen3/gcc-11.2.0/hdf5-1.12.2-pcmpothn4sir2fls465cojwxkkwrvdxq/lib/libhdf5_fortran.so /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-gnu/spack/opt/spack/cray-sles15-zen3/gcc-11.2.0/hdf5-1.12.2-pcmpothn4sir2fls465cojwxkkwrvdxq/lib/libhdf5_f90cstub.so /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-gnu/spack/opt/spack/cray-sles15-zen3/gcc-11.2.0/hdf5-1.12.2-pcmpothn4sir2fls465cojwxkkwrvdxq/lib/libhdf5.so' '--with-zlib=1' '--with-zlib-include=/gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-gnu/spack/opt/spack/cray-sles15-zen3/gcc-11.2.0/zlib-1.2.12-ymk6b64y6cp65why5fpmer6joa6enbzj/include' '--with-zlib-lib=/gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-gnu/spack/opt/spack/cray-sles15-zen3/gcc-11.2.0/zlib-1.2.12-ymk6b64y6cp65why5fpmer6joa6enbzj/lib/libz.so' '--with-mumps=0' '--with-trilinos=0' '--with-fftw=0' '--with-valgrind=0' '--with-gmp=0' '--with-libpng=0' '--with-giflib=0' '--with-mpfr=0' '--with-netcdf=0' '--with-pnetcdf=0' '--with-moab=0' '--with-random123=0' '--with-exodusii=0' '--with-cgns=0' '--with-memkind=0' '--with-p4est=0' '--with-saws=0' '--with-yaml=0' '--with-hwloc=0' '--with-libjpeg=0' '--with-scalapack=0' '--with-strumpack=0' '--with-mmg=0' '--with-parmmg=0' '--with-tetgen=0' '--with-hip-arch=gfx90a' 'HIPPPFLAGS=-I/opt/rocm-5.2.0/include -I/opt/rocm-5.2.0/include -I/opt/rocm-5.2.0/include -I/opt/rocm-5.2.0/include -I/opt/rocm-5.2.0/include -I/opt/rocm-5.2.0/include -I/opt/rocm-5.2.0/include -I/opt/rocm-5.2.0/include ' 'with-hip-lib=/opt/rocm-5.2.0/lib/libhipsparse.so /opt/rocm-5.2.0/lib/libhipblas.so /opt/rocm-5.2.0/lib/librocsparse.so /opt/rocm-5.2.0/lib/librocsolver.so /opt/rocm-5.2.0/lib/librocblas.so  -L/opt/rocm-5.2.0/hip/lib -lamdhip64'

4 warnings found in build log:
     72    TESTING: checkDynamicLinker from config.setCompilers(/tmp/lpeyrala/spack-stage/spack-stage-petsc-3.17.4-z3h5sxpo2kegc6eghkrcxsgyqrbsxrvr/spack-src/config/BuildSyste
           m/config/setCompilers.py:2440)
     73    TESTING: checkPragma from config.setCompilers(/tmp/lpeyrala/spack-stage/spack-stage-petsc-3.17.4-z3h5sxpo2kegc6eghkrcxsgyqrbsxrvr/spack-src/config/BuildSystem/confi
           g/setCompilers.py:1900)
     74    TESTING: output from config.setCompilers(/tmp/lpeyrala/spack-stage/spack-stage-petsc-3.17.4-z3h5sxpo2kegc6eghkrcxsgyqrbsxrvr/spack-src/config/BuildSystem/config/set
           Compilers.py:2489)
     75    TESTING: configureIndexSize from PETSc.options.indexTypes(/tmp/lpeyrala/spack-stage/spack-stage-petsc-3.17.4-z3h5sxpo2kegc6eghkrcxsgyqrbsxrvr/spack-src/config/PETSc
           /options/indexTypes.py:39)
     76    TESTING: configureCompilerFlags from config.compilerFlags(/tmp/lpeyrala/spack-stage/spack-stage-petsc-3.17.4-z3h5sxpo2kegc6eghkrcxsgyqrbsxrvr/spack-src/config/Build
           System/config/compilerFlags.py:82)
     77                      =============================================================================================
  >> 78    ***** WARNING: Using default optimization C flags -g -O
     79                            You might consider manually setting optimal optimization flags for your system with
     80                             COPTFLAGS="optimization flags" see config/examples/arch-*-opt.py for examples
     81                      =============================================================================================
     82
     83                      =============================================================================================
  >> 84                            ***** WARNING: Using default Cxx optimization flags -g -O
     85                            You might consider manually setting optimal optimization flags for your system with
     86                             CXXOPTFLAGS="optimization flags" see config/examples/arch-*-opt.py for examples
     87                      =============================================================================================
     88
     89                      =============================================================================================
  >> 90                            ***** WARNING: Using default FORTRAN optimization flags -g -O
     91                            You might consider manually setting optimal optimization flags for your system with
     92                             FOPTFLAGS="optimization flags" see config/examples/arch-*-opt.py for examples
     93                      =============================================================================================
     94
     95                      =============================================================================================
  >> 96                            ***** WARNING: Using default HIP optimization flags -g -O3
     97                            You might consider manually setting optimal optimization flags for your system with
     98                             HIPOPTFLAGS="optimization flags" see config/examples/arch-*-opt.py for examples
     99                      =============================================================================================
     100
     101   TESTING: checkIntelHardwareSupport from config.compilerFlags(/tmp/lpeyrala/spack-stage/spack-stage-petsc-3.17.4-z3h5sxpo2kegc6eghkrcxsgyqrbsxrvr/spack-src/config/Bu
           ildSystem/config/compilerFlags.py:173)
     102   TESTING: checkCompilerMacros from config.compilerFlags(/tmp/lpeyrala/spack-stage/spack-stage-petsc-3.17.4-z3h5sxpo2kegc6eghkrcxsgyqrbsxrvr/spack-src/config/BuildSys
           tem/config/compilerFlags.py:164)
```
[spack-build-out.txt](https://github.com/spack/spack/files/9648649/spack-build-out.txt)
[spack-build-env.txt](https://github.com/spack/spack/files/9648650/spack-build-env.txt)

### Error message

<details><summary>Error message</summary><pre>
... see above
</pre></details>

### Information on your system

* **Spack:** 0.19.0.dev0 (fce58626df2a4f526de0771adfa8b7afb673a361)
* **Python:** 3.9.7
* **Platform:** cray-sles15-zen3
* **Concretizer:** clingo

### Additional information

[configure.log](https://github.com/spack/spack/files/9648653/configure.log)
[spack-build-out.txt](https://github.com/spack/spack/files/9648658/spack-build-out.txt)
[spack-build-env.txt](https://github.com/spack/spack/files/9648659/spack-build-env.txt)

@balay @BarrySmith @jedbrown @wspear @lukebroskop 

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

### PR DESCRIPTION
**Pull request by eugeneswalker**
Fixes https://github.com/spack/spack/issues/32826 based on https://github.com/spack/spack/issues/32826#issuecomment-1276547306

FYI @balay @renjithravindrankannath @srekolam @wspear

### CODE REVIEW

File: var/spack/repos/builtin/packages/hip/package.py
Comment (balay) [inline var/spack/repos/builtin/packages/hip/package.py]: I didn't update this comment in my inital patch - perhaps:
```suggestion
            # We assume self.spec.prefix is  /opt/rocm-x.y.z for rocm-5.2.0 and newer
            # and /opt/rocm-x.y.z/hip for older versions
```

**Comment by tldahlgren** [comment]
`ERROR: Job failed: pod "runner-xqazbqsk-project-2-concurrent-4bldpr" status is "Failed"`

@spackbot run pipeline

**Comment by eugeneswalker** [comment]
All checks are passing here. Let us give some time for `hip` maintainers to weigh in before merging.

**Review by balay — APPROVED** [review]
This is more like a self-approve

**Comment by balay** [comment]
I see a related MR https://github.com/spack/spack/pull/33302 - but don't think there are any conflicts with this change

**Pull request merged**
